ethereum/forge-test/Bridge.t.sol

```diff
@@ -20,26 +31,105 @@ contract ExportedBridge is Bridge {
+    uint8 constant finality = 15;
+
+    // "TokenBridge" (left padded)
+    bytes32 constant tokenBridgeModule =
+        0x000000000000000000000000000000000000000000546f6b656e427269646765;
+    uint8 actionRegisterChain = 1;
+    uint8 actionContractUpgrade = 2;
+    uint8 actionRecoverChainId = 3;
+
+    uint16 fakeChainId = 1337;
+    uint256 fakeEvmChainId = 10001;
+
+    uint16 testForeignChainId = 1;
+    bytes32 testForeignBridgeContract =
+        0x0000000000000000000000000000000000000000000000000000000000000004;
+    uint16 testBridgedAssetChain = 1;
+    bytes32 testBridgedAssetAddress =
+        0x000000000000000000000000b7a2211e8165943192ad04f5dd21bedc29ff003e;
+
+    uint256 public constant testGuardian =
+        93941733246223705020089879371323733820373732307041878556247502674739205313440;
 
     function setUp() public {
-        bridge = new ExportedBridge();
+        // Setup wormhole
+        implementationTest = new TestImplementation();
+        implementationTest.setUp();
+
+        // Get wormhole from implementation tests
+        wormhole = IWormhole(address(implementationTest.proxied()));
+
+        // Deploy setup
+        bridgeSetup = new BridgeSetup();
+        // Deploy implementation contract
+        bridgeImpl = new ExportedBridge();
+        // Deploy token implementation
+        tokenImpl = new TokenImplementation();
+        // Deploy WETH
+        weth = IERC20(address(new MockWETH9()));
+
+        testChainId = implementationTest.testChainId();
+        testEvmChainId = implementationTest.testEvmChainId();
+        vm.chainId(testEvmChainId);
+        governanceChainId = implementationTest.governanceChainId();
+        governanceContract = implementationTest.governanceContract();
+
+        bytes memory setupAbi = abi.encodeWithSelector(
+            BridgeSetup.setup.selector,
+            address(bridgeImpl),
+            testChainId,
+            address(wormhole),
+            governanceChainId,
+            governanceContract,
+            address(tokenImpl),
+            address(weth),
+            finality,
+            testEvmChainId
+        );
+
+        // Deploy proxy
+        bridge = ITokenBridgeTest(
+            address(new TokenBridge(address(bridgeSetup), setupAbi))
+        );
     }
 
     function testTruncate(bytes32 b) public {
         bool invalidAddress = bytes12(b) != 0;
         if (invalidAddress) {
-            vm.expectRevert( "invalid EVM address");
+            vm.expectRevert("invalid EVM address");
         }
-        bytes32 converted = bytes32(uint256(uint160(bytes20(bridge._truncateAddressPub(b)))));
+        bytes32 converted = bytes32(
+            uint256(uint160(bytes20(bridge._truncateAddressPub(b))))
+        );
 
         if (!invalidAddress) {
             require(converted == b, "truncate does not roundrip");
         }
     }
 
-    function testEvmChainId() public {
+    function testSetEvmChainId() public {
         vm.chainId(1);
```

Comment:
This test should use the variables that are set in state (e.g., `fakeChainId`).

Reply:
Agree this should be addressed with #4090